pkg/logs/client/destination.go

```diff
@@ -5,8 +5,10 @@
 
 package client
 
+import "github.com/DataDog/datadog-agent/pkg/logs/message"
+
 // Destination sends a payload to a specific endpoint over a given network protocol.
 type Destination interface {
-	Send(payload []byte) error
-	SendAsync(payload []byte)
+	Start(input chan *message.Payload, output chan *message.Payload) (stopChan chan struct{})
```

Comment:
I think we want to return a directional write-only chan to the caller for isolation:
```suggestion
	Start(input chan *message.Payload, output chan *message.Payload) (stopChan chan<- struct{})
```

As a follow-up, I think enforcing directional channels in other parts of the changes in the PR may be beneficial too.

Reply:
makes sense.